src/main/java/org/ohdsi/analysis/featureextraction/design/CovariateSettings.java

```diff
@@ -0,0 +1,352 @@
+package org.ohdsi.analysis.featureextraction.design;
+
+import com.fasterxml.jackson.annotation.JsonGetter;
+import java.util.List;
+import org.ohdsi.analysis.RLangClass;
+
+public interface CovariateSettings extends RLangClass {
+  @JsonGetter("temporal")
+  Boolean getTemporal();
+
+  @JsonGetter("DemographicsGender")
+  Boolean getDemographicsGender();
```

Comment:
Do we really need to list all the options as separate fields? Cannot we deal with Map?

Reply:
Leaving these for now since it makes things easier when comparing to the FeatureExtraction library. Plus, this object represents the function signature for creating covariate settings in that package.